pkg/object/s3.go

```diff
@@ -306,39 +323,46 @@ func newS3(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 		ep         string
 	)
 
-	if len(hostParts) == 1 { // take endpoint as bucketname
+	if len(hostParts) == 1 {
```

Comment:
It could be a valid hostname in path-style, for example, http://localhost/bucket

Reply:
Added